### PR TITLE
feature/add-sphinx-docs

### PR DESCRIPTION
**Pull request by PabloCorcho**
This PR includes a new directory containing sphinx documentation

### CODE REVIEW

File: docs/source/get_ssp_data.rst
Comment (paranoya) [inline docs/source/get_ssp_data.rst:1]: Is this file referenced anywhere? Shouldn't it appear in `index.rst`?

File: docs/source/api.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+Observables
+-----------
+.. automodule:: pst.observables
+   :members:
+   :undoc-members:
+   :show-inheritance:
+
+Fitting
+-----------
```
Comment (paranoya) [inline docs/source/api.rst:33]: Too many "-"?

File: docs/source/conf.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    ]
+    sys.modules.update((mod_name, MagicMock()) for mod_name in MOCK_MODULES)
+
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'Population Synthesis Toolkit'
+copyright = '2024, Ascasibar Y., Corcho-Caballero P., Jiménez-López D.'
+author = 'Ascasibar Y., Corcho-Caballero P., Jiménez-López D.'
```
Comment (paranoya) [inline docs/source/conf.py:53]: No sure about the author ordering, unless it is merely alphabetic

**Review by paranoya — APPROVED** [review]

File: docs/source/get_ssp_data.rst
Comment (PabloCorcho) [reply docs/source/get_ssp_data.rst:1]: My bad, I missed to push one commit.

File: docs/source/api.rst (hunk 1 above)
Comment (PabloCorcho) [reply docs/source/api.rst:33]: The number of dashes is irrelevant provided that it covers the entire word. We can make the source code prettier in future PR.

File: docs/source/conf.py (hunk 2 above)
Comment (PabloCorcho) [reply docs/source/conf.py:53]: Yes, I put the names in alphabetical order

**Comment by PabloCorcho** [comment]
@paranoya we can now upload the docs to reathedocs.

**Pull request merged**